### ISSUE TITLE
Paddle 2.2 Layer.to API always reports warnings

### ISSUE THREAD

**Issue by Grant-Tao** [PaddlePaddle/Paddle#38453]
**System information**
Aistudio python 3.7 paddle 2.2.1

Layer.to function sample code :

code :
import paddle

linear = paddle.nn.Linear(3,4)
linear.to('cpu')

reporting: 
xxx\ paddle\fluid\dygraph\varbase_patch_methods.py:392: UserWarning: 
Warning: tensor.grad will return the tensor value of the gradient. This is an incompatible upgrade for tensor.grad API.  It's return type changes from numpy.ndarray in version 2.0 to paddle.Tensor in version 2.1.0.  If you want to get the numpy value of the gradient, you can use :code:`x.grad.numpy()` 
 warnings.warn(warning_msg)
in ParamBase copy_to func
in ParamBase copy_to func

======
Is this normal? When I used this api  for my large model, error reported.

btw, is there a convenient api for model transfering between cpu and gpu in paddle?

**Comment by yingyibiao** [comment]
Thanks for your question! You can use paddle.device.set_device API to set the global device. It may not be a good practice in paddle to transfer model between different devices. When not specified to a CPU device, GPU device will be used with higher priority whenever it's available.

**Comment by Grant-Tao** [comment]
@yingyibiao thanks a lot. If I have two models, (as GPU can only accomodate one model), one in CPU memory, one in GPU memory, I need exchange them periodically, how could I do? creating a new model on GPU each time, then set state dict?  
and what is the layer.to api used for?
